pandas/_libs/writers.pyx

```diff
@@ -127,19 +127,28 @@ def max_len_string_array(pandas_string[:] arr) -> Py_ssize_t:
 
     for i in range(length):
         val = arr[i]
-        if isinstance(val, str):
-            l = PyString_GET_SIZE(val)
-        elif isinstance(val, bytes):
-            l = PyBytes_GET_SIZE(val)
-        elif isinstance(val, unicode):
-            l = PyUnicode_GET_SIZE(val)
+        l = max_len_string(val)
 
         if l > m:
             m = l
 
     return m
 
 
+cpdef inline Py_ssize_t max_len_string(object val):
+    """ return the maximum size of a string"""
+    cdef:
+        Py_ssize_t l = 0
+
+    if isinstance(val, str):
+        l = PyString_GET_SIZE(val)
+    elif isinstance(val, bytes):
+        l = PyBytes_GET_SIZE(val)
+    elif isinstance(val, unicode):
```

Comment:
Don't need this check any more as we dropped Py2 support

Reply:
Done! Thanks!